src/d3a/models/state.py

```diff
@@ -237,6 +336,19 @@ class PVState(ProductionState):
     Completely inherits ProductionState, but we keep this class for backward compatibility.
     """
 
+    def _calculate_unsettled_energy_kWh(
+            self, measured_energy_kWh: float, time_slot: DateTime) -> float:
+        """
+        Returns negative values for overproduction (offer will be placed on the settlement market)
+        and positive values for underproduction (bid will be placed on the settlement market)
+        :param measured_energy_kWh: Measured energy that the PV produced
+        :param time_slot: time slot of the measured energy
+        :return: Deviation between forecasted and measured energy
+        """
+        traded_energy_kWh = (self.get_energy_production_forecast_kWh(time_slot) -
+                             self.get_available_energy_kWh(time_slot))
+        return traded_energy_kWh - measured_energy_kWh
```

Comment:
```suggestion
        return self.get_energy_production_forecast_kWh(time_slot) - measured_energy_kWh
```
Shouldn't it be the difference of forecasted and measured energy?

Reply:
No, you need to substract the traded energy, too. Please have a look at the docstring.

Reply:
Not really. This measures the deviation of the measurements from the traded energy. The way it goes: 
spot market -> takes forecast as input, and decides production / consumption based on traded energy. This is the energy that the asset has to produce / consume.
settlement market -> the measurements indicate a deviation of production/ consumption from the energy that the asset decided to sell / buy. Therefore it uses this deviation in order to palce bids and offers to the settlement market.

Reply:
> No, you need to substract the traded energy, too. Please have a look at the docstring

Left a [comment](https://github.com/gridsingularity/d3a/pull/1213/files?file-filters%5B%5D=.py#r699407475) to update docstring accordingly.